ui/src/tree/AzExtTreeDataProvider.ts

```diff
@@ -40,11 +40,11 @@ export class AzExtTreeDataProvider implements IAzExtTreeDataProviderInternal, ty
         return this._onTreeItemCreateEmitter.event;
     }
 
-    public getTreeItem(treeItem: AzExtTreeItem): TreeItem {
+    public getTreeItem(treeItem: AzExtTreeItem & { uniqueFullId?: string }): TreeItem {
         return {
             label: treeItem.label,
             description: treeItem.effectiveDescription,
-            id: treeItem.fullId,
+            id: treeItem.uniqueFullId || treeItem.fullId,
```

Comment:
I'd prefer having `uniqueFullId` just being it's own property on treeItems that need it.  And then using it in the `getChildren` commmands that need it.  

I don't like replacing id with this uniqueFullId even if you have to pass it in.  I think it also makes the `getChildren` command look really awkward and clunky.

Reply:
Actually just ignore my comment.  I reread Eric's comment and this seems like what he suggested 🤔

Reply:
Well, but I do agree the `getChildren` method looks clunky now. Maybe see if you can pull the "duplicate" logic out into a separate method

Reply:
> I'd prefer having uniqueFullId just being it's own property on treeItems that need it. And then using it in the getChildren commands that need it.

Not sure if this is what Nathan is saying, but I'm okay with making `uniqueFullId` a property on `AzExtTreeItem` itself - as long as you only do it in `AzExtTreeItem.ts` and _not_ `index.d.ts` (which would expose the property outside the ui package).

Reply:
That wasn't exactly what I meant, but I saw your comment about not exposing it outside the ui package and I agree.  I think this is what I'd prefer too ^